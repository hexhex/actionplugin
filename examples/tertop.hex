#maxint = 6.

mul1(X,Y) :- 6 = X * Y, #int(X), #int(Y).
mul2(X,Y) :- *(X, Y, 6), #int(X), #int(Y).
plus1(X,Y) :- 6=X+Y, #int(X), #int(Y).
plus2(X,Y) :- +(X,Y,6), #int(X), #int(Y).

