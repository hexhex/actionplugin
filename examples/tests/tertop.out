{plus2(0,6), plus2(1,5), plus2(2,4), plus2(3,3), plus2(4,2), plus2(5,1), plus2(6,0), plus1(0,6), plus1(1,5), plus1(2,4), plus1(3,3), plus1(4,2), plus1(5,1), plus1(6,0), mul2(1,6), mul2(2,3), mul2(3,2), mul2(6,1), mul1(1,6), mul1(2,3), mul1(3,2), mul1(6,1)}
